### ISSUE TITLE
Filter $plan_id in the `[lifterlms_checkout]` shortcode.

### ISSUE THREAD

**Issue by jasonyingling** [gocodebox/lifterlms#1471]
**Is your feature request related to a problem? Please describe.**
I'm trying to have the homepage of my site display the user checkout form for a specific free Membership access plan. Since the site will only have a single plan.

The `filter_input` function used by `llms_filter_input` does not use the current value of `$_GET['plan']`. So when a user lands on page with no query parameters in the URL `filter_input` always returns `null`. Even if I explicitly set `$_GET['plan'] = 335;` to match my access plan id. So `$plan_id` on line 190 of `class.llms.shortcode.checkout.php` always fails.

**Describe the solution you'd like**
I need to be able to filter the `$plan_id`value on line 190 of `class.llms.shortcode.checkout.php`.

Something like:

```
$plan_id = llms_filter_input( INPUT_GET, 'plan', FILTER_SANITIZE_NUMBER_INT );

$plan_id = apply_filters( 'llms_filter_checkout_plan_id', $plan_id );
```

**Describe alternatives you've considered**
Is there another way to immediately show the checkout form?

**Comment by thomasplevy** [comment]
@jasonyingling 

Have you tried adding the filter yourself already and ensuring this will work?

I haven't looked at much of this code in a while but a quick glance leaves me to believe that after getting the form to display without a query string variable you'll start to encounter other issues.

I'm pretty sure the shortcode won't really function that well outside of the checkout page (where it's intended to be used). There's some short-sighted architectural issues at play behind the scenes here.

We've had requests to allow the shortcode to be used in any context (not just on the checkout page) and I'd like to make that happen some day but it's not on our immediate internal roadmap as a priority at this moment.

**Comment by jasonyingling** [comment]
@thomasplevy

I did have it functioning with that filter in place for my specific use case. But I wasn't sure how widely this use case would be reasonable and if it would introduce additional bugs.

Thinking through my case some more and poking around the Lifter code base, I decided it was probably a more reasonable route for me to go to use the registration form instead and enroll the student in the membership using the LLMS_Student::enroll` hooking into the registration hooks that are available.

That allowed me to accomplish the problem I was trying to solve. So I'd say this issue is safe to close.
